src/deadline/job_attachments/download.py

```diff
@@ -372,48 +372,25 @@ def download_files_in_directory(
     )
 
 
-def download_file(
-    file: RelativeFilePath,
-    hash_algorithm: HashAlgorithm,
-    local_download_dir: str,
+def download_file_with_s3_key(
     s3_bucket: str,
-    cas_prefix: Optional[str],
-    s3_client: Optional[BaseClient] = None,
+    s3_key: str,
+    transfer_manager,
+    local_file_name: Path,
+    file_bytes: Optional[int] = 0,
     session: Optional[boto3.Session] = None,
-    modified_time_override: Optional[float] = None,
     progress_tracker: Optional[ProgressTracker] = None,
     file_conflict_resolution: Optional[FileConflictResolution] = FileConflictResolution.CREATE_COPY,
-) -> Tuple[int, Optional[Path]]:
+) -> concurrent.futures.Future | None:
```

Comment:
Instead of using ` | None` we generally use `Optional[concurrent.futures.Future]` instead.